packages/app/src/runner/studio/AssertionType.ce.vue

```diff
@@ -0,0 +1,119 @@
+<template>
+  <div
+    :class="['assertion-type', { 'single-assertion': !hasOptions }]"
+    @click="_click"
+    @mouseover="_open"
+    @mouseout="_close"
+  >
+    <div class="assertion-type-text">
+      <span>
+        {{ type.replace(/\./g, ' ') }}
+      </span>
+      <span
+        v-if="hasOptions"
+        class="dropdown-arrow"
+      >
+        <svg
+          xmlns="http://www.w3.org/2000/svg"
+          width="12"
+          height="12"
+          fill="currentColor"
+          viewBox="0 0 16 16"
+        >
+          <path
+            fillRule="evenodd"
+            d="M4.646 1.646a.5.5 0 0 1 .708 0l6 6a.5.5 0 0 1 0 .708l-6 6a.5.5 0 0 1-.708-.708L10.293 8 4.646 2.354a.5.5 0 0 1 0-.708z"
+          />
+        </svg>
+      </span>
+    </div>
+    <AssertionOptions
+      v-if="hasOptions && isOpen"
+      :type="type"
+      :add-assertion="addAssertion"
+      :options="options"
+      :set-popper-element="_setPopperElement"
+    />
+  </div>
+</template>
+
+<script lang="ts" setup>
+import { Ref, ref } from 'vue'
+import AssertionOptions from './AssertionOptions.ce.vue'
+
+const props = defineProps<{
+  type: string
+  addAssertion: any
+  options: any
+}>()
+
+const isOpen = ref(false)
+const hasOptions = props.options && !!props.options.length
+const popperElement: Ref<HTMLElement | null> = ref(null)
+
+const _open = (e) => {
+  isOpen.value = true
+  e.stopPropagation()
+}
+
+const _close = (e) => {
+  if (popperElement.value && popperElement.value.contains(e.relatedTarget)) {
+    return
+  }
+
+  isOpen.value = false
+  e.stopPropagation()
```

Comment:
might be a good opportunity to leverage [event modifiers](https://vuejs.org/guide/essentials/event-handling.html#event-modifiers) to bake into the template over the handler and would eliminate the need to pass in the event

Reply:
Done.